lib/ethon/easy/queryable.rb

```diff
@@ -95,18 +96,37 @@ def mime_type(filename)
       def recursively_generate_pairs(h, prefix, pairs)
         case h
         when Hash
-          h.each_pair do |k,v|
-            key = prefix.nil? ? k : "#{prefix}[#{k}]"
-            pairs_for(v, key, pairs)
-          end
+          encode_hash_pairs(h, prefix, pairs)
         when Array
-          h.each_with_index do |v, i|
-            key = "#{prefix}[#{i}]"
-            pairs_for(v, key, pairs)
+          if rack_arrays
+            encode_rack_array_pairs(h, prefix, pairs)
+          else
+            encode_indexed_array_pairs(h, prefix, pairs)
           end
         end
       end
 
+      def encode_hash_pairs(h, prefix, pairs)
+        h.each_pair do |k,v|
+          key = prefix.nil? ? k : "#{prefix}[#{k}]"
+          pairs_for(v, key, pairs)
+        end
+      end
+
+      def encode_indexed_array_pairs(h, prefix, pairs)
+        h.each_with_index do |v, i|
+          key = "#{prefix}[#{i}]"
+          pairs_for(v, key, pairs)
+        end
+      end
+
+      def encode_rack_array_pairs(h, prefix, pairs)
```

Comment:
you implemented rack style encoding here, but there is something provided by Rack: http://www.rubydoc.info/github/rack/rack/master/Rack/Utils#build_query-class_method. What do you think about taking that one?